src/util/metadata/vo_bit/mod.rs

```diff
@@ -98,45 +98,36 @@ pub fn is_vo_bit_set<VM: VMBinding>(object: ObjectReference) -> bool {
 }
 
 /// Check if an address can be turned directly into an object reference using the VO bit.
-/// If so, return `Some(object)`. Otherwise return `None`.
-pub fn is_vo_bit_set_for_addr<VM: VMBinding>(address: Address) -> Option<ObjectReference> {
-    let potential_object = ObjectReference::from_raw_address(address)?;
-
-    let addr = potential_object.to_address::<VM>();
-
-    // If we haven't mapped VO bit for the address, it cannot be an object
-    if !VO_BIT_SIDE_METADATA_SPEC.is_mapped(addr) {
-        return None;
-    }
-
-    if VO_BIT_SIDE_METADATA_SPEC.load_atomic::<u8>(addr, Ordering::SeqCst) == 1 {
-        Some(potential_object)
-    } else {
-        None
-    }
+pub fn is_vo_bit_set_for_addr<VM: VMBinding>(address: Address) -> bool {
```

Comment:
This function used to return `ObjectReference`. However we cannot be sure that the `ObjectReference` is actually valid. I just change it to return boolean, and let the caller figure out the object reference.